File: 프로그래머스/unrated/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.js
function solution(arr, k) {
    arr = new Set(arr);
    arr = Array.from(arr);
    
    if (arr.length < k) {
        for (let i = arr.length; i <= k; i++) {
            arr.push(-1);
        }
    }
    
    return arr.slice(0, k);
}

File: 프로그래머스/unrated/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        Set<Integer> set = new HashSet<>();
        int[] result = new int[k];
        Arrays.fill(result, -1);
        
        int i = 0;
        for (int num : arr) {
            if (i >= k) {
                return result;
            }
            
            if (!set.contains(num)) {
                result[i++] = num;
                set.add(num);
            }
        }
        
        return result;
    }
}